### PR TITLE
Correct computation of angular momentum

### PR DESCRIPTION
**Pull request by cphyc**
## PR Summary

Until now, there was a mistake in the computation of angular momentum which was leading to the angular momentum units being messed up.

## PR Checklist

- [x] Code passes flake8 checker
- [x] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

File: yt/fields/tests/test_angular_momentum.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#-----------------------------------------------------------------------------
+
+import numpy as np
+
+import yt
+from yt.testing import \
+    assert_allclose_units, \
+    requires_file
+
+@requires_file('output_00080/info_00080.txt')
```
Comment (ngoldbaum) [inline yt/fields/tests/test_angular_momentum.py]: Any chance we can run this test using e.g. `fake_amr_ds` or some other dataset that doesn't need to download. I don't think anything about this test is specific to this RAMSES output?

**Comment by ngoldbaum** [comment]
The test failures on python3 are unrelated to this PR.

File: yt/fields/tests/test_angular_momentum.py (hunk 1 above)
Comment (cphyc) [reply yt/fields/tests/test_angular_momentum.py]: I'm not sure why but the test passes even without the fix when using `fake_amr_ds`. Any guess why?

**Comment by cphyc** [comment]
Both were wrong. This is essentially due to the centre being expressed in the wrong units (hence the relative position was wrong, etc.).

**Comment by cphyc** [comment]
Both were wrong. This is essentially due to the centre being expressed in the wrong units (hence the relative position was wrong, etc.).

**Comment by chummels** [comment]
That's odd.  I've definitely used this in the past and it has worked to get the angular momentum vector for a galactic disk.  I never used the magnitude of the vector for anything, but the direction seemed to be spot on.  Also, I would've thought the flurry of PRs regarding angular momentum in yt-4 a few months back would've caught this issue: PR #1902 PR #1934 .

**Comment by cphyc** [comment]
The bug is triggered in the case where units mismatch, so I guess that if your datasets were using the "right" units, you didn't run into the bug.

**Comment by chummels** [comment]
Oh I see, that makes sense...  Well, either way--great job!

**Pull request merged**